networks/suzuka/suzuka-full-node/src/partial.rs

```diff
@@ -1,57 +1,77 @@
-use std::{sync::Arc, time::Duration};
-
-use anyhow::Context;
+use crate::SuzukaFullNode;
+use m1_da_light_node_client::{
+	blob_response, BatchWriteRequest, BlobWrite, LightNodeServiceClient,
+	StreamReadFromHeightRequest,
+};
+use mcr_settlement_client::{mock::MockMcrSettlementClient, McrSettlementClientOperations};
+use mcr_settlement_manager::{
+	CommitmentEventStream, McrSettlementManager, McrSettlementManagerOperations,
+};
+use movement_types::{Block, BlockCommitmentEvent};
 use suzuka_executor::{
-    SuzukaExecutor,
-    ExecutableBlock,
-    HashValue,
-    FinalityMode,
-    Transaction,
-    SignatureVerifiedTransaction,
-    SignedTransaction,
-    ExecutableTransactions,
-    v1::SuzukaExecutorV1,
+	v1::SuzukaExecutorV1, ExecutableBlock, ExecutableTransactions, FinalityMode, HashValue,
+	SignatureVerifiedTransaction, SignedTransaction, SuzukaExecutor, Transaction,
 };
-use m1_da_light_node_client::*;
-use async_channel::{Sender, Receiver};
+
+use anyhow::Context;
+use async_channel::{Receiver, Sender};
 use sha2::Digest;
-use crate::*;
-use tokio_stream::StreamExt;
 use tokio::sync::RwLock;
-use movement_types::Block;
+use tokio_stream::StreamExt;
 
+use std::future::Future;
+use std::sync::Arc;
+use std::time::Duration;
 
-#[derive(Clone)]
-pub struct SuzukaPartialNode<T : SuzukaExecutor + Send + Sync + Clone> {
-    executor: T,
-    transaction_sender : Sender<SignedTransaction>,
-    pub transaction_receiver : Receiver<SignedTransaction>,
-    light_node_client: Arc<RwLock<LightNodeServiceClient<tonic::transport::Channel>>>,
+pub struct SuzukaPartialNode<T> {
```

Comment:
Let's add in a doc comment for the struct and all its fields.

Reply:
There should be a wider ticket for documenting our public APIs (when they are stabilized enough to avoid unnecessary churn). I have filed #81 to track this.